distributed/utils_test.py

```diff
@@ -974,7 +974,7 @@ def coro():
                                 if s.validate:
                                     s.validate_state()
                             finally:
-                                if client:
+                                if client and c.status != "closed":
```

Comment:
Probably worth checking, but there might also be a `closing` status. 
```suggestion
                                if client and not c.status.startswith("clos"):
```